### PR TITLE
chore(Upload): improve typing of `id` property

### PR DESCRIPTION
**Pull request by langz**

### CODE REVIEW

File: packages/dnb-eufemia/src/components/upload/types.ts [hunk 1]
```diff
@@ -16,7 +16,12 @@ export type UploadProps = {
   /**
    * unique id used with the useUpload hook to manage the files
    */
-  id: string
+  id:
+    | string
+    | (() => void)
+    | Promise<() => void>
+    | React.Context<any>
+    | Record<string, unknown>
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/upload/types.ts:24]:
```suggestion
  id: SharedStateId
```

NB: The import is missing.

**Review by tujoworker — APPROVED** [review]
Nice. I think we can used/import the this type `SharedStateId` for simplification. We do that other places too.

File: packages/dnb-eufemia/src/components/upload/types.ts (hunk 1 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/upload/types.ts:24]: Yeah, of course. 
Not sure why I did not do it, I think I thought it was from Forms, but it's clearly not 👍

Comment (langz) [reply packages/dnb-eufemia/src/components/upload/types.ts]: Updated now :)

**Pull request merged**